library/general/src/modules/Popup.rb

```diff
@@ -250,31 +250,19 @@ def NoHeadline
     #
     # @return [Yast::Term] button box
     def AnyQuestionButtonBox(yes_button_message, no_button_message, focus)
-      yes_button = Empty()
-      no_button = Empty()
+      yes_opts = [:okButton]
+      no_opts = [:cancelButton]
 
       if focus == :focus_no
-        yes_button = PushButton(Id(:yes), Opt(:okButton), yes_button_message)
-        no_button = PushButton(
-          Id(:no_button),
-          Opt(:default, :cancelButton),
-          no_button_message
-        )
+        no_opts << :default
       else
-        yes_button = PushButton(
-          Id(:yes),
-          Opt(:default, :okButton),
-          yes_button_message
-        )
-        no_button = PushButton(
-          Id(:no_button),
-          Opt(:cancelButton),
-          no_button_message
-        )
+        yes_opts << :default
       end
 
-      button_box = ButtonBox(yes_button, no_button)
-      deep_copy(button_box)
+      yes_button = PushButton(Id(:yes), Opt(*yes_opts), yes_button_message)
+      no_button = PushButton(Id(:no), Opt(*no_opts), no_button_message)
+
+      ButtonBox(yes_button, no_button)
```

Comment:
it a bit scary me to refactor it without test coverage in https://github.com/yast/yast-yast2/blob/master/library/general/test/popup_test.rb but if you test it enough manually then it has green from me

Reply:
Tested manually, works fine.